pkg/refactor/engine/descent-info.go

```diff
@@ -42,6 +42,23 @@ func (info *InstanceDescentInfo) MarshalJSON() ([]byte, error) {
 	})
 }
 
+func (info *InstanceDescentInfo) Callpath() string {
```

Comment:
I am not sure if this is the right place for these functions. Everywhere InstanceDescentInfo is passed, so is InstanceTelemetryInfo, which already has the callpath. We should minimize the number of places complicated concepts/logic like callpath is defined.